### ISSUE TITLE
Mod for announcements bar?

### ISSUE THREAD

**Issue by pablopeu** [cccraig/FancyFooter#2]
Hi CC, just saw your plugin listed, before its existence I agonized hacking something similar for my site (www.peu.net) Im not a coder :) 
I wonder if you think it would be useful to implement an announcements bar on top of the page. As a free client of Cloudflare, they have an app for doing this, but I think a native piwigo plugin could serve us better.
Thanks!!

Here is how my site looks with the cloudflare app:
![image](https://user-images.githubusercontent.com/25783916/34579383-92b9321c-f167-11e7-8663-949ba3b30686.png)

**Comment by cccraig** [comment]
Sure I'll take a look at making a notification bar plugin. It might take me a little time to do so, I was able to do my other piwigo plugins at work because we were creating a self hosted photo gallery. This shouldn't be terribly complicated to implement so I'll let you know when I have a working demo.

**Comment by pablopeu** [comment]
Great!!! let me add some sugestions for customization and food for thought:

Colors: bar color/text color/button color/button text color
Behaviour: auto close upon set timer and/or close by pressing the X
Cookie based behaviour: show every visit/once a day/once period of time
Size: bar height/button height
Alignment: text L/C/R/Justified

what else... :)

add icon/jpg to bar or button?
configure multiple messages? carroussel/random appearance

Thanks!!

**Comment by cccraig** [comment]
Yes I think these are good ideas. I'm not sure about the carroussel
notification bar, I think something like that would drive me nuts on a
website. I think I will also add a link to whatever it is being updated.
For instance, (using your site as an example) if you make a new knife and
post a gallery there would be a link to the new gallery. I *might* be able
to do that automatically which would be cool.

I had some big projects come in all at once at work but all thats under
control now so i'm going to try to get at least a beta up on the piwigo
site this weekend or maybe early next week.

**Comment by pablopeu** [comment]
Great, count me in for testing!

**Comment by cccraig** [comment]
I've got a working version on my github now,
https://github.com/cccraig/PiwiBar. There are still a few additions to be
made but It's working! Right now the notification bar shows based on
whether your message changed or if the url changed but I plan to add a
frequency option and the rest of the options mentioned earlier.

I have included an option for "test mode". This will always show the bar
for an administrator/webmaster but should keep the expected behavior for
other users.

**Comment by pablopeu** [comment]
Excelent! I don't see it yet in the piwigo extensions page and don't know
how to download it straight from github, I'm sure its a matter of hours.
Will install it ASAP and let you know if I notice quirks or bugs.
THANKS A TON!

**Comment by cccraig** [comment]
Hi! I have the plugin on Piwigo now!

http://piwigo.org/ext/extension_view.php?eid=873&rid=6519#rev6519


On Tue, Jan 23, 2018 at 10:52 AM, Pablo E. Untroib <notifications@github.com

**Comment by pablopeu** [comment]
Great! but I'm on holidays (it's summer here) I'm anxious to check it out!
